pkg/compose/convergence.go

```diff
@@ -126,17 +126,22 @@ func updateServices(service *types.ServiceConfig, cnts Containers) {
 	if len(cnts) == 0 {
 		return
 	}
-	cnt := cnts[0]
-	serviceName := cnt.Labels[api.ServiceLabel]
+	serviceName2containerID := make(map[string]string)
+	for _, cnt := range cnts {
+		serviceName := cnt.Labels[api.ServiceLabel]
+		if _, exists := serviceName2containerID[serviceName]; !exists {
+			serviceName2containerID[serviceName] = cnt.ID
+		}
+	}
 
-	if d := getDependentServiceFromMode(service.NetworkMode); d == serviceName {
-		service.NetworkMode = types.NetworkModeContainerPrefix + cnt.ID
```

Comment:
you could use `cnts.filter(isService(d))` to retrieve project containers matching the requested service

Reply:
Thank you for the advice!